### PR TITLE
tests: copy autoscaler e2e test

### PR DESCRIPTION
**Pull request by paulfantom**
e2e smoke tests

/CC @enxebre @frobware

### CODE REVIEW

File: cluster-autoscaler/test/e2e/main.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+const (
+	namespace = "openshift-cluster-api"
+)
+
+func init() {
+	if err := capiv1alpha1.AddToScheme(scheme.Scheme); err != nil {
+		glog.Fatal(err)
+	}
+
+	if err := osconfigv1.AddToScheme(scheme.Scheme); err != nil {
```
Comment (enxebre) [inline cluster-autoscaler/test/e2e/main.go]: I don't think this is being used at all?

File: cluster-autoscaler/test/e2e/main.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
+package main
```
Comment (enxebre) [inline cluster-autoscaler/test/e2e/main.go]: we might want to put this into upper level folder e.g `root/openshift/test/e2e` so it does not interfere with dep manager for `autoscaler/` binary

File: cluster-autoscaler/Makefile [hunk 3]
```diff
@@ -21,7 +21,7 @@ test-unit: clean deps build
 	$(ENVVAR) godep go test --test.short -race ./... $(FLAGS)
 
 test-e2e:
-	$(ENVVAR) godep go run ./test/e2e/*.go -alsologtostderr
+	cd test && go run ./e2e/*.go -alsologtostderr
```
Comment (enxebre) [inline cluster-autoscaler/Makefile]: can you either squash into the previews commit modifying make file or having all make changes into a single atomic commit?

File: cluster-autoscaler/test/e2e/main.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+}
+
+func runSuite() error {
+
+	client, err := newClient()
+	if err != nil {
+		return err
+	}
+	testConfig := &testConfig{
+		client: client,
```
Comment (enxebre) [inline cluster-autoscaler/test/e2e/main.go]: Either here on in a follow up we could run https://github.com/openshift/cluster-autoscaler-operator/blob/master/test/e2e/operator_expectations.go#L59 for robustness

File: cluster-autoscaler/test/e2e/main.go (hunk 2 above)
Comment (ingvagabund) [reply cluster-autoscaler/test/e2e/main.go]: I would even prefer to move it one dir up. Even though the repo carries more components, only the cluster-autoscaler is actually utilized.

File: cluster-autoscaler/Makefile (hunk 3 above)
Comment (ingvagabund) [reply cluster-autoscaler/Makefile]: Please merge the first and the last commit into one.

File: cluster-autoscaler/test/e2e/main.go (hunk 2 above)
Comment (paulfantom) [reply cluster-autoscaler/test/e2e/main.go]: done, tests live in `test`

File: test/e2e/main.go [hunk 5]
```diff
@@ -0,0 +1,73 @@
+package main
```
Comment (enxebre) [inline test/e2e/main.go]: I'd scope this to `test/openshift/e2e` @frobware @ingvagabund wdyt?

**Comment by frobware** [comment]
Yeah. I think anything which is related to OpenShift should convey that
fact. Less surprises as to what things are when we rebase to later
versions.

File: test/e2e/main.go (hunk 5 above)
Comment (ingvagabund) [reply test/e2e/main.go]: sgtm

**Comment by paulfantom** [comment]
PR enabling this test in CI: https://github.com/openshift/release/pull/2664

**Pull request merged**